test/Microsoft.AspNet.Mvc.Razor.Test/RazorViewEngineTest.cs

```diff
@@ -345,6 +348,227 @@ public void FindView_UsesAreaViewLocationFormat_IfRouteContainsArea()
         }
 
         [Theory]
+        [InlineData("Test-View.cshtml")]
+        [InlineData("/Home/Test-View.cshtml")]
+        public void FindView_DoesNotUseViewLocationFormat_WithRelativePath_IfRouteDoesNotContainArea(string viewName)
+        {
+            // Arrange
+            var expectedViewName = "/Home/Test-View.cshtml";
+            var pageFactory = new Mock<IRazorPageFactory>();
+            var viewFactory = new Mock<IRazorViewFactory>();
+            var page = Mock.Of<IRazorPage>();
+            pageFactory
+                .Setup(p => p.CreateInstance(expectedViewName))
+                .Returns(page)
+                .Verifiable();
+
+            var viewEngine = new OverloadedLocationViewEngine(
+                pageFactory.Object,
+                viewFactory.Object,
+                GetOptionsAccessor(),
+                GetViewLocationCache());
+            var actionContext = GetActionContext(_controllerTestContext);
+            var context = new ViewContext(
+                actionContext,
+                Mock.Of<IView>(),
+                new ViewDataDictionary(new EmptyModelMetadataProvider()),
+                Mock.Of<ITempDataDictionary>(),
+                TextWriter.Null,
+                new HtmlHelperOptions())
+            {
+                ExecutingFilePath = "/Home/Page.cshtml",
+            };
+
+            viewFactory
+                .Setup(v => v.GetView(viewEngine, page, false))
+                .Returns(Mock.Of<IView>());
+
+            // Act
+            var result = viewEngine.FindView(context, viewName);
+
+            // Assert
+            Assert.True(result.Success);
+            Assert.Equal(viewName, result.ViewName);    // Result gets original view name.
```

Comment:
Getting the original view name is not a behaviour change. `RazorViewEngine` does this consistently whether it uses view locations or not.